### PR TITLE
Add .lycheeignore and changelogs/fragments to workflow ignore

### LINKED ISSUE #6772: [chore] do not run workflows for lycheeignore changes
Anytime there is a link failure (which seems to be an increasing amount of times lately). We need to update the lycheeignore file if the link is unrelated to the actual application.

When there is update, the PR runs the full workflow. Building and testing. We have a set number of runners which can block runs on other PRs.

We should not run those workflows on lycheeignore changes.

Update for merge and pull request here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.github/workflows/build_and_test_workflow.yml#L15

And here: https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/.github/workflows/cypress_workflow.yml#L8

### PR DESCRIPTION
**Pull request by sejli**
### Description
Adds `.lycheeignore` and `changelogs/fragments/**` to skip `build_and_test` and `cypress` workflows.

### Issues Resolved
#6772 

## Screenshot

N/A

## Testing the changes

N/A

## Changelog
- chore: Add .lycheeignore and changelogs/fragments to workflow ignore

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: .github/workflows/build_and_test_workflow.yml [hunk 1]
```diff
@@ -15,6 +15,8 @@ on:
     paths-ignore:
       - '**/*.md'
       - 'docs/**'
+      - '.lycheeignore'
+      - 'changelogs/fragments/**'
```
Comment (manasvinibs) [inline .github/workflows/build_and_test_workflow.yml:20]: Hi @sejli, don't we have to add the paths ignore for Push/merge trigger as well?

**Comment by BionIT** [comment]
Hi @sejli , are we targeting to get this PR into 2.15?

**Comment by sejli** [comment]
Hi @BionIT, this PR should be targeting `2.15`.

**Comment by BionIT** [comment]
> Hi @BionIT, this PR should be targeting `2.15`.

Could you address the comments from @manasvinibs ?

**Review by manasvinibs — APPROVED** [review]

**Review by BionIT — APPROVED** [review]

**Pull request merged**